scrapy_google_sheets_exporter/gsheets_exporter.py

```diff
@@ -0,0 +1,98 @@
+import csv
+import logging
+from io import TextIOWrapper
+from urllib.parse import parse_qs, urlparse
+
+import gspread
+from gspread.exceptions import NoValidUrlKeyFound
+from scrapy.exceptions import NotConfigured
+from scrapy.extensions.feedexport import BlockingFeedStorage, build_storage
+
+logger = logging.getLogger(__name__)
+
+
+class GoogleSheetsFeedStorage(BlockingFeedStorage):
+    def __init__(self, uri, credentials, *, feed_options=None):
+
+        self.feed_options = feed_options or {}
+        self.overwrite = self.feed_options.get("overwrite", False)
+        self.format = self.feed_options.get("format", "csv")
+        self.batch_size = self.feed_options.get("batch_size", 0)
```

Comment:
I think we might want to use a different name here, to avoid confusion (and potential forward conflicts) with the file-based batching feature. `stream_batch_size` maybe?

I also wonder if we need this setting at all, given how the implementation in this initial version needs to wait for the whole crawl to finish and be written to disk first. Maybe we could consider mentioning the write-to-disk-first implementation detail in the README to avoid surprises, and remove this option altogether, keeping only the size-based criteria to handle stream batching, and not even mentioning that in the docs, since it is an implementation detail in the end and we are following the recommended practices.